injector/injector.go

```diff
@@ -350,8 +350,8 @@ func (i *SecretInjector) InjectSecretsFromVaultPath(paths string, inject SecretI
 
 		version := "-1"
 
-		if len(split) > 2 {
-			version = split[2]
+		if len(split) == 2 {
+			version = split[1]
```

Comment:
I'm a little bit confused here. Given the string `secret/data/account#password#1` the two chunks will be `secret/data/account` and `password#1`.

Am I missing something?

Reply:
That is a good question, but according to the [docs](https://bank-vaults.dev/docs/mutating-webhook/annotations/), we expect a "_Comma-delimited list of vault paths to pull in **all secrets** as environment variables_" in the `vault.security.banzaicloud.io/vault-env-from-path` annotation, and this list is what the method in question is about. So I concluded that the intention here was to use it like this: `secret/data/account#1`. This is not stated explicitly in the docs, so if we are okay with this behaviour, I thought of submitting a PR to the docs repo as well to clarify this.

Reply:
This makes sense by the way because if a pod needs all of the secrets from a secret path, this annotation provides an easier alternative to adding them as envs one by one...

Reply:
Fair enough, but then I'm not sure if this PR fixes the original issue (or maybe I misunderstood what that issue was about)

Reply:
The confusion hides in `InjectSecretsFromVault` and `InjectSecretsFromVaultPath` since they do not work in the same way. AFAIK the latter is "pseudo-subset" of the first one. We should definitely document this behaviour. Making them work the same way will break the API, so this should resolve the issue without affecting the API.

Reply:
For me, this looks good since it does resolve issue with handling the versioning in `InjectSecretsFromVaultPath`. @akijakya verified this by running the tests without the fix.

Reply:
Yes, `InjectSecretsFromVault` injects the singular values from one or more `secrets/path#name`, `InjectSecretsFromVaultPath` injects all secrets from one or more `secrets/path`. Maybe the names could be better, or we could include this clarification as comment...

Reply:
A comment with an example of operation for both methods would be good. But we definitely need this transparent across the docs as well.